src/plugins/wip/index.js

```diff
@@ -0,0 +1,184 @@
+    });
+}
+
+/**
+ * Check to see if there is an existing pending wip status check on the PR.
+ * If so, create a success wip status check.
+ * @param {Object} context Probot context object
+ * @param {string} sha Commit SHA hash associated with the status check
+ * @returns {Promise} A Promise which will resolve when a success status check
+ * is created on the PR, or an immediately-resolved Promise if no status check
+ * is needed.
+ */
+async function maybeResolveWipStatusOnPR(context, sha) {
+    const repoAndRef = context.repo({
+        ref: sha
+    });
+
+    let statusCheckExists = false;
+
+    await context.github.paginate(
+        context.github.repos.getCombinedStatusForRef(repoAndRef),
+        (res, done) => {
+            for (const status of res.data.statuses) {
+                if (status.context === "wip") {
+                    statusCheckExists = true;
+                    done();
+                }
+            }
+        }
+    );
+
+    if (statusCheckExists) {
+        return createSuccessWipStatusOnPR(context, sha);
+    }
+
+    return null;
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context Probot context object
+ * @param {Object} pr pull request object from GitHub's API
+ * @returns {Promise<Object[]>} A Promise that fulfills with a list of commit objects from GitHub's API
+ * @private
+ */
+async function getAllCommitsForPR({ context, pr }) {
+    const { data: commitList } = await context.github.pullRequests.getCommits(
+        context.repo({ number: pr.number })
+    );
+
+    return commitList;
+}
+
+/**
+ * Checks to see if a PR has the "do not merge" label.
+ * @param {Array<Object>} labels - collection of label objects
+ * @returns {boolean} True if release label is present
+ * @private
+ */
+function hasDoNotMergeLabel(labels) {
+    return labels.some(({ name }) => name === DO_NOT_MERGE_LABEL);
+}
+
+/**
+ * Get the sha value from the latest commit
+ * @param {Object[]} allCommits A list of commit objects from GitHub's API
+ * @returns {string} latest commit sha
+ * @private
+ */
+function pluckLatestCommitSha(allCommits) {
+    return allCommits[allCommits.length - 1].sha;
+}
+
+/**
+ * Checks to see if a PR's title has a WIP indication.
+ * @param {Object} pr Pull request object
+ * @returns {boolean} True if the PR title indicates WIP
+ * @private
+ */
+function prHasWipTitle(pr) {
+    return WIP_IN_TITLE_REGEX.test(pr.title);
+}
+
+/**
+ * Handler for PR events (opened, reopened, synchronized, edited, labeled,
+ * unlabeled).
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function prChangedHandler(context) {
+    const pr = context.payload.pull_request;
+
+    const allCommits = await getAllCommitsForPR({
+        context,
+        pr
+    });
+
+    const sha = pluckLatestCommitSha(allCommits);
+
+    const prTitleHasWip = prHasWipTitle(pr);
+    const hasWipLabel = hasDoNotMergeLabel(pr.labels);
```

Comment:
nit: Since function names are readable, you dont need this and the one below variable.